src/config/singletons.js

```diff
@@ -1,20 +1,28 @@
-let singletonsConfig;
+/* global web3 */
 
-const defaultGas = 6e6;
-const defaultSender = '';
+let singletonsConfig;
 
 function setSingletonsConfig (config) {
   singletonsConfig = config;
 }
 
 function setDefaultSingletonsConfig () {
-  setSingletonsConfig({ defaultGas, defaultSender });
+  setSingletonsConfig({
+    abstraction: isTruffleEnvironment() ? 'truffle' : 'web3',
+    defaultGas: 8e6,
+    defaultSender: '',
```

Comment:
What is the effect of this?

Reply:
It means that all transactions sent using that contract object require an explicit `from` field